# AirBnB NY Locations Data Case Study

In this final project, you task will be to take the data provided and find evidence to answer the following questions.

1. Which hosts are the busiest and why?
2. How many neighborhood groups are available and which shows up the most?
3. Are private rooms the most popular in manhattan?
4. Which hosts are the busiest and based on their reviews?
5. Which neighorhood group has the highest average price?
6. Which neighborhood group has the highest total price?
7. Which top 5 hosts have the highest total price?
8. Who currently has no (zero) availability with a review count of 100 or more?
9. What host has the highest total of prices and where are they located?
10. When did Danielle from Queens last receive a review?


In [3]:
import numpy as np
import pandas as pd



1. Which hosts are busiest and why?

In [4]:
df = pd.read_csv(r'C:\Users\bioni\OneDrive\Documents\GitHub\Coding_Temple\Week 3\Assignments\Wkend Project\AB_NYC_2019.csv')

df.columns

Index(['id', 'name', 'host_id', 'host_name', 'neighbourhood_group',
       'neighbourhood', 'latitude', 'longitude', 'room_type', 'price',
       'minimum_nights', 'number_of_reviews', 'last_review',
       'reviews_per_month', 'calculated_host_listings_count',
       'availability_365'],
      dtype='object')

In [5]:
df.groupby(['host_name']).count().sort_values('availability_365', ascending = False)[['calculated_host_listings_count','availability_365','reviews_per_month']]


calculated_host_listings_count  availability_365  \
host_name                                                        
Michael                                  417               417   
David                                    403               403   
Sonder (NYC)                             327               327   
John                                     294               294   
Alex                                     279               279   
...                                      ...               ...   
Jerbean                                    1                 1   
Jerald                                     1                 1   
Jeonghoon                                  1                 1   
Jeny                                       1                 1   
현선                                         1                 1   

              reviews_per_month  
host_name                        
Michael                     335  
David                       309  
Sonder (NYC)                207  
John                        250  
Alex                        229  
...                         ...  
Jerbean                       1  
Jerald                        1  
Jeonghoon                     0  
Jeny                          1  
현선                            0  

[11452 rows x 3 columns]

The hosts with the most properties and highest reviews are typically the busiest, they are also among the most with no availability. john was the busiest because he had the most amount of properties and reviews per month.


In [6]:
# How many neighborhood groups are available and which shows up the most?

df.groupby(['neighbourhood_group']).count().sort_values(['neighbourhood'], ascending=False)

id   name  host_id  host_name  neighbourhood  \
neighbourhood_group                                                    
Manhattan            21661  21652    21661      21652          21661   
Brooklyn             20104  20098    20104      20095          20104   
Queens                5666   5666     5666       5664           5666   
Bronx                 1091   1090     1091       1090           1091   
Staten Island          373    373      373        373            373   

                     latitude  longitude  room_type  price  minimum_nights  \
neighbourhood_group                                                          
Manhattan               21661      21661      21661  21661           21661   
Brooklyn                20104      20104      20104  20104           20104   
Queens                   5666       5666       5666   5666            5666   
Bronx                    1091       1091       1091   1091            1091   
Staten Island             373        373        373    373             373   

                     number_of_reviews  last_review  reviews_per_month  \
neighbourhood_group                                                      
Manhattan                        21661        16632              16632   
Brooklyn                         20104        16447              16447   
Queens                            5666         4574               4574   
Bronx                             1091          876                876   
Staten Island                      373          314                314   

                     calculated_host_listings_count  availability_365  
neighbourhood_group                                                    
Manhattan                                     21661             21661  
Brooklyn                                      20104             20104  
Queens                                         5666              5666  
Bronx                                          1091              1091  
Staten Island                                   373               373

Manhatten and Brooklyn have, by far, the most listings by count. Manhatten tops the charts with the most listings overall.

In [7]:
# Are private rooms the most popular in manhattan?

df.groupby(['neighbourhood_group','room_type',]).count().sort_values('availability_365', ascending= False)[['calculated_host_listings_count']]

calculated_host_listings_count
neighbourhood_group room_type                                      
Manhattan           Entire home/apt                           13199
Brooklyn            Private room                              10132
                    Entire home/apt                            9559
Manhattan           Private room                               7982
Queens              Private room                               3372
                    Entire home/apt                            2096
Bronx               Private room                                652
Manhattan           Shared room                                 480
Brooklyn            Shared room                                 413
Bronx               Entire home/apt                             379
Queens              Shared room                                 198
Staten Island       Private room                                188
                    Entire home/apt                             176
Bronx               Shared room                                  60
Staten Island       Shared room                                   9

Based on our review, entire home/apt are the most popular in Manhatten with 13,199 listings.
Private rooms were second in popularity with 7,982.

In [8]:
# Which hosts are the busiest and based on their reviews?

df.groupby(['host_name']).count().sort_values('reviews_per_month', ascending = False)[['calculated_host_listings_count','availability_365','reviews_per_month']]


calculated_host_listings_count  availability_365  \
host_name                                                        
Michael                                  417               417   
David                                    403               403   
John                                     294               294   
Alex                                     279               279   
Sonder (NYC)                             327               327   
...                                      ...               ...   
Joscelyn                                   1                 1   
Dae                                        1                 1   
Navarra                                    1                 1   
Dagmara                                    1                 1   
현선                                         1                 1   

              reviews_per_month  
host_name                        
Michael                     335  
David                       309  
John                        250  
Alex                        229  
Sonder (NYC)                207  
...                         ...  
Joscelyn                      0  
Dae                           0  
Navarra                       0  
Dagmara                       0  
현선                            0  

[11452 rows x 3 columns]

Based on our review; David, Michael, Alex, and Sonder were the hosts that were the busiest based on thier calculated host rating and reviews per month. Overall, John had the highest occupancy rate per property, per review each month.

In [9]:
#Which neighorhood group has the highest average price?
df.groupby(['neighbourhood_group']).mean(True).sort_values(['price'],ascending = False)[['price']].round(2)

price
neighbourhood_group        
Manhattan            196.88
Brooklyn             124.38
Staten Island        114.81
Queens                99.52
Bronx                 87.50

Manhatten had the highest average price at $196.88 per night.

In [19]:
# Which neighbor hood group has the highest total price?
df.groupby(['neighbourhood_group']).max(True).sort_values(['price'],ascending = False)[['price']]

price
neighbourhood_group       
Brooklyn             10000
Manhattan            10000
Queens               10000
Staten Island         5000
Bronx                 2500

Manhatten had the highest total price for all listings combined. However, it is interesting to note that; Manhatten, Queens, and Brooklyn all had listings that entertained $10,000 per night listings. 

In [11]:
#Which top 5 hosts have the highest total price?
df.groupby(['host_name']).max(True).sort_values(['price'],ascending = False)[['price']].head(5)

price
host_name       
Kathrine   10000
Erin       10000
Jelena     10000
Amy         9999
Olson       9999

In [12]:
# Who currently has no (zero) availability with a review count of 100 or more?
conditional = (df['number_of_reviews'] >= 100) & (df['availability_365']==0)
df[conditional].groupby(['host_name']).count().sort_values(['number_of_reviews'], ascending=False)[['number_of_reviews','availability_365']]



number_of_reviews  availability_365
host_name                                           
Michael                          5                 5
Gurpreet  Singh                  4                 4
Brian                            3                 3
Emily And Joel                   2                 2
Jo & Savannah                    2                 2
...                            ...               ...
Ehren                            1                 1
Edward                           1                 1
Dragan                           1                 1
Doug                             1                 1
正川                               1                 1

[141 rows x 2 columns]

141 had zero availability with a review count of 100 or more. 

In [13]:
# What host has the highest total of prices and where are they located?

df.groupby(['neighbourhood_group','neighbourhood','host_name']).max(numeric_only=True).sort_values(['price'], ascending = False)[['price']].head(1)

price
neighbourhood_group neighbourhood host_name       
Queens              Astoria       Kathrine   10000

In [14]:
# When did Danielle from Queens last receive a review?

df[df['host_name']=='Danielle'].groupby(['last_review']).max(False).sort_values(['number_of_reviews'],ascending = False).head(5)



id                                               name  \
last_review                                                                
7/6/2019     22686497                  Sun Soaked & Cozy PR Williamsburg   
7/8/2019     18173787  Cute Tiny Room Family Home by LGA NO CLEANING FEE   
7/3/2019      5115372  Comfy Room Family Home LGA Airport NO CLEANING...   
7/7/2019     17222454   Sun Room Family Home LGA Airport NO CLEANING FEE   
6/25/2019     1807106                   Stunning  Limestone  1 Bdrm Flat   

              host_id host_name neighbourhood_group  neighbourhood  latitude  \
last_review                                                                    
7/6/2019     26432133  Danielle              Queens   Williamsburg  40.76389   
7/8/2019     26432133  Danielle              Queens  East Elmhurst  40.76380   
7/3/2019     26432133  Danielle              Queens  East Elmhurst  40.76374   
7/7/2019     26432133  Danielle              Queens  East Elmhurst  40.76367   
6/25/2019     9223085  Danielle            Brooklyn  Crown Heights  40.67191   

             longitude        room_type  price  minimum_nights  \
last_review                                                      
7/6/2019     -73.87007      Shared room    130               4   
7/8/2019     -73.87238     Private room     48               1   
7/3/2019     -73.87103     Private room     54               1   
7/7/2019     -73.87088     Private room     48               1   
6/25/2019    -73.93997  Entire home/apt    135               3   

             number_of_reviews  reviews_per_month  \
last_review                                         
7/6/2019                   510              16.22   
7/8/2019                   436              16.03   
7/3/2019                   430              13.45   
7/7/2019                   417              14.36   
6/25/2019                  190               2.77   

             calculated_host_listings_count  availability_365  
last_review                                                    
7/6/2019                                  5               354  
7/8/2019                                  5               337  
7/3/2019                                  5               347  
7/7/2019                                  5               338  
6/25/2019                                 1               267

7/8/2019

## Further Questions

1. Which host has the most listings?

In [15]:
df.groupby(['host_name']).count().sort_values(['calculated_host_listings_count'], ascending=False)[['calculated_host_listings_count']]

calculated_host_listings_count
host_name                                   
Michael                                  417
David                                    403
Sonder (NYC)                             327
John                                     294
Alex                                     279
...                                      ...
Jerbean                                    1
Jerald                                     1
Jeonghoon                                  1
Jeny                                       1
현선                                         1

[11452 rows x 1 columns]

Michael has the most listing per count at 417.

2. How many listings have completely open availability?

In [16]:
df.groupby(['host_name']).count().sort_values('availability_365', ascending = False)[['availability_365']].tail(6903)

availability_365
host_name                       
Nataraj                        1
Milo                           1
Naheel                         1
Natanel                        1
Milly & George                 1
...                          ...
Jerbean                        1
Jerald                         1
Jeonghoon                      1
Jeny                           1
현선                             1

[6903 rows x 1 columns]

6903

3. What room_types have the highest review numbers?

In [26]:
#df.groupby(['room_type']).max().sort_values('number_of_reviews', ascending=False)[['number_of_reviews','reviews_per_month']]
#df.groupby(['room_type']).mean(True).sort_values('reviews_per_month', ascending=True)[['reviews_per_month']]
df.groupby(['room_type',]).count().sort_values('reviews_per_month', ascending=False)[['reviews_per_month']]

reviews_per_month
room_type                         
Entire home/apt              20332
Private room                 17665
Shared room                    846

Entire home/apt led the total reviews per month with 20,332 and with total overall number of reviews at 25,409.

# Final Conclusion

In this cell, write your final conclusion for each of the questions asked.

Also, if you uncovered some more details that were not asked above, please describe them here.

-- Add your conclusion --

Final anlysis:
The hosts with the most total amount of listings tended to have higher occupancy and correlating higher review ratings and count. Manhatten led all markets with the most amount of listings, total possible revenue opportunity, least amount of open vancancy, highest per night fee, and total house/apartment rentals. It is intersting to note; there were quite a few open vacancies for the single listing hosts, this reinforces the theory of smaller hosts do not fill vacancy as easily as hosts with numerous properties.